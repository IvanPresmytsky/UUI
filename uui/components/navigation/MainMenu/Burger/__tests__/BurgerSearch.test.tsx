import React from 'react';
import { BurgerSearch } from '../BurgerSearch';
import { ReactComponent as CalendarIcon } from '../../../../../icons/calendar-24.svg';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('BurgerSearch', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<BurgerSearch value={ null } onValueChange={ () => {} } />);

        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <BurgerSearch value={ null } onValueChange={ () => {} } onAccept={ () => {} } onCancel={ () => {} } icon={ CalendarIcon } iconPosition="right" isDropdown isOpen />,
        );

        expect(tree).toMatchSnapshot();
    });
});
