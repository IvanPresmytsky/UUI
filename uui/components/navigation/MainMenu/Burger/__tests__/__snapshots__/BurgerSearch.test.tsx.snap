// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerSearch should be rendered correctly 1`] = `
<div
  className="container uui-input-box -clickable searchInput uui-main_menu-burger-search"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
  <input
    autoFocus={true}
    className="uui-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="text"
    value=""
  />
</div>
`;

exports[`BurgerSearch should be rendered correctly 2`] = `
<div
  className="container uui-input-box -clickable searchInput uui-main_menu-burger-search"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
  <input
    autoFocus={true}
    className="uui-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="text"
    value=""
  />
</div>
`;
